core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -483,7 +483,9 @@ public void close(ScheduledExecutorService instance) {
    */
   public static ThreadFactory getThreadFactory(String nameFormat, boolean daemon) {
     if (IS_RESTRICTED_APPENGINE) {
-      return MoreExecutors.platformThreadFactory();
+      @SuppressWarnings("BetaApi")
+      ThreadFactory factory = MoreExecutors.platformThreadFactory();
```

Comment:
idea: in the future we could reflectively find this so that future versions of guava work with older versions of gRPC.